### ISSUE TITLE
TypeLoadException in Telemetry assembly causes dev14 with new Roslyn bits to crash 

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#12072]
Repro:
- Install dev14 (I'm using update 3 preview specifically)
- Build Roslyn (open only)
- Launch dev hive and try to open a solution like Roslyn.sln

Symptom:
The IDE crashes with following information.

`{"Method 'AnalyzeSyntaxAsync' in type 'Analyzer' from assembly 'Microsoft.VisualStudio.LanguageServices.Telemetry, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.":"Analyzer"}`

```
   at Microsoft.VisualStudio.LangaugeServices.Telemetry.ProjectTelemetryIncrementalAnalyzerProvider.CreateIncrementalAnalyzer(Workspace workspace)
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.<>c__DisplayClass15_0.<GetOrderedAnalyzers>b__0(Lazy`2 p) in D:\repos\roslyn2\src\Features\Core\Portable\SolutionCrawler\WorkCoordinator.IncrementalAnalyzerProcessor.cs:line 100
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at System.Collections.Immutable.ImmutableArray.CreateRange[T](IEnumerable`1 items)
   at System.Collections.Immutable.ImmutableArray.ToImmutableArray[TSource](IEnumerable`1 items)
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.GetOrderedAnalyzers(Registration registration, IEnumerable`1 providers) in D:\repos\roslyn2\src\Features\Core\Portable\SolutionCrawler\WorkCoordinator.IncrementalAnalyzerProcessor.cs:line 100
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.GetIncrementalAnalyzers(Registration registration, IEnumerable`1 providers) in D:\repos\roslyn2\src\Features\Core\Portable\SolutionCrawler\WorkCoordinator.IncrementalAnalyzerProcessor.cs:line 90
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.<>c__DisplayClass11_0.<.ctor>b__1() in D:\repos\roslyn2\src\Features\Core\Portable\SolutionCrawler\WorkCoordinator.IncrementalAnalyzerProcessor.cs:line 59
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.NormalPriorityProcessor.get_Analyzers() in D:\repos\roslyn2\src\Features\Core\Portable\SolutionCrawler\WorkCoordinator.NormalPriorityProcessor.cs:line 71
   at Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.NormalPriorityProcessor.<ResetStatesAsync>d__42.MoveNext() in D:\repos\roslyn2\src\Features\Core\Portable\SolutionCrawler\WorkCoordinator.NormalPriorityProcessor.cs:line 535
```

This exception occurs because of a change to IIncrementalAsync interface (in open): https://github.com/dotnet/roslyn/commit/52d00a2fabe7b43381e8ab077ff7f6c3e8733e69 

CC @KevinH-MS

**Comment by Pilchie** [comment]
Tag @heejaechang

**Comment by KevinH-MS** [comment]
This is due to a "cross-cutting" change that was made for **features/test-impact**.  We'll get this reverted in **roslyn-internal/master** and **roslyn/master** today.

Lesson is that we can't make any breaking, cross-cutting changes (even internal ones) as long as the **master** branch can deploy to multiple versions of VS.

**Comment by KevinH-MS** [comment]
Fixed with #12074
